### ISSUE TITLE
command:espIdf.getToolchainGdb undefined (VSC-1363)

### ISSUE THREAD

**Issue by marcinczenko** [espressif/vscode-esp-idf-extension#1203]
### OS

MacOS

### Operating System version

macOS 14.5

### Visual Studio Code version

1.89.1

### ESP-IDF version

v5.4.0

### Python version

3.12.3

### Doctor command output

---------------------------------------------- ESP-IDF Extension for Visual Studio Code report ---------------------------------------------
OS darwin arm64 23.5.0 
System environment variable IDF_PYTHON_ENV_PATH 
 undefined 
System environment variable PATH 
 /var/folders/ns/r3r4k_9j39q42bmfjh5s7pwh0000gn/T/frum_2333_1716516253621/bin:<HOMEPATH>/.pyenv/shims:<HOMEPATH>/micro:<HOMEPATH>/.bun/bin:<HOMEPATH>/.nvm/versions/node/v20.13.1/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/Library/Apple/usr/bin:/Applications/Keybase.app/Contents/SharedSupport/bin:<HOMEPATH>/.cargo/bin:<HOMEPATH>/go/bin:<HOMEPATH>/.rvm/bin 
System environment variable PYTHON 
 undefined 
Visual Studio Code version 1.89.1 
Visual Studio Code language en 
Visual Studio Code shell /bin/zsh 
ESP-IDF Extension version 1.7.1 
Workspace folder <HOMEPATH>/code/embedded/native-esp-idf/example-native-esp-idf 
---------------------------------------------------- Extension configuration settings ------------------------------------------------------
ESP-ADF Path (idf.espAdfPath) ${env:ADF_PATH}
ESP-IDF Path (idf.espIdfPath) <HOMEPATH>/code/embedded/esp/esp-idf
ESP-MDF Path (idf.espMdfPath) ${env:MDF_PATH}
ESP-Matter Path (idf.espMatterPath) ${env:ESP_MATTER_PATH}
ESP-HomeKit-SDK Path (idf.espHomeKitSdkPath) ${env:HOMEKIT_PATH}
Custom extra paths (idf.customExtraPaths) <HOMEPATH>/.espressif/tools/xtensa-esp-elf-gdb/14.2_20240403/xtensa-esp-elf-gdb/bin:<HOMEPATH>/.espressif/tools/riscv32-esp-elf-gdb/14.2_20240403/riscv32-esp-elf-gdb/bin:<HOMEPATH>/.espressif/tools/xtensa-esp-elf/esp-13.2.0_20240305/xtensa-esp-elf/bin:<HOMEPATH>/.espressif/tools/riscv32-esp-elf/esp-13.2.0_20240305/riscv32-esp-elf/bin:<HOMEPATH>/.espressif/tools/esp32ulp-elf/2.38_20240113/esp32ulp-elf/bin:<HOMEPATH>/.espressif/tools/openocd-esp32/v0.12.0-esp32-20240318/openocd-esp32/bin:<HOMEPATH>/.espressif/tools/esp-rom-elfs/20240305
Custom extra vars (idf.customExtraVars)
    OPENOCD_SCRIPTS: <HOMEPATH>/.espressif/tools/openocd-esp32/v0.12.0-esp32-20240318/openocd-esp32/share/openocd/scripts
    ESP_ROM_ELF_DIR: <HOMEPATH>/.espressif/tools/esp-rom-elfs/20240305/
Virtual env Python Path (idf.pythonBinPath) <HOMEPATH>/.espressif/python_env/idf5.4_py3.12_env/bin/python
Serial port (idf.port) /dev/cu.usbserial-0001
OpenOCD Configs (idf.openOcdConfigs) interface/ftdi/olimex-arm-usb-tiny-h.cfg,board/esp-wroom-32.cfg
ESP-IDF Tools Path (idf.toolsPath) <HOMEPATH>/.espressif
Git Path (idf.gitPath) git
Notification Mode (idf.notificationMode) All
-------------------------------------------------------- Configurations access -------------------------------------------------------------
Access to ESP-ADF Path (idf.espAdfPath) false
Access to ESP-IDF Path (idf.espIdfPath) true
Access to ESP-MDF Path (idf.espMdfPath) false
Access to ESP-Matter Path (idf.espMatterPath) false
Access to ESP-HomeKit Path (idf.espHomeKitSdkPath) false
Access to ESP-IDF Custom extra paths
Access to <HOMEPATH>/.espressif/tools/xtensa-esp-elf-gdb/14.2_20240403/xtensa-esp-elf-gdb/bin: true
Access to <HOMEPATH>/.espressif/tools/riscv32-esp-elf-gdb/14.2_20240403/riscv32-esp-elf-gdb/bin: true
Access to <HOMEPATH>/.espressif/tools/xtensa-esp-elf/esp-13.2.0_20240305/xtensa-esp-elf/bin: true
Access to <HOMEPATH>/.espressif/tools/riscv32-esp-elf/esp-13.2.0_20240305/riscv32-esp-elf/bin: true
Access to <HOMEPATH>/.espressif/tools/esp32ulp-elf/2.38_20240113/esp32ulp-elf/bin: true
Access to <HOMEPATH>/.espressif/tools/openocd-esp32/v0.12.0-esp32-20240318/openocd-esp32/bin: true
Access to <HOMEPATH>/.espressif/tools/esp-rom-elfs/20240305: true
Access to Virtual env Python Path (idf.pythonBinPath) true
Access to CMake in environment PATH true
Access to Ninja in environment PATH true
Access to ESP-IDF Tools Path (idf.toolsPath) true
-------------------------------------------------------- Configurations has spaces -------------------------------------------------------------
Spaces in system environment Path false
Spaces in ESP-ADF Path (idf.espAdfPath) false
Spaces in ESP-IDF Path (idf.espIdfPath) false
Spaces in ESP-MDF Path (idf.espMdfPath) false
Spaces in ESP-Matter Path (idf.espMatterPath) false
Spaces in ESP-HomeKit-SDK Path (idf.espHomeKitSdkPath) false
Spaces in ESP-IDF Custom extra paths
Spaces in <HOMEPATH>/.espressif/tools/xtensa-esp-elf-gdb/14.2_20240403/xtensa-esp-elf-gdb/bin: false
Spaces in <HOMEPATH>/.espressif/tools/riscv32-esp-elf-gdb/14.2_20240403/riscv32-esp-elf-gdb/bin: false
Spaces in <HOMEPATH>/.espressif/tools/xtensa-esp-elf/esp-13.2.0_20240305/xtensa-esp-elf/bin: false
Spaces in <HOMEPATH>/.espressif/tools/riscv32-esp-elf/esp-13.2.0_20240305/riscv32-esp-elf/bin: false
Spaces in <HOMEPATH>/.espressif/tools/esp32ulp-elf/2.38_20240113/esp32ulp-elf/bin: false
Spaces in <HOMEPATH>/.espressif/tools/openocd-esp32/v0.12.0-esp32-20240318/openocd-esp32/bin: false
Spaces in <HOMEPATH>/.espressif/tools/esp-rom-elfs/20240305: false
Spaces in Virtual env Python Path (idf.pythonBinPath) false
Spaces in ESP-IDF Tools Path (idf.toolsPath) false
----------------------------------------------------------- Executables Versions -----------------------------------------------------------
Git version 2.45.1
ESP-IDF version 5.4.0
Python version 3.12.3
Python's pip version 24.0
-------------------------------------------------- Project configuration settings ----------------------------------------------------------
-------------------------------------------------- Python packages in idf.pythonBinPath ----------------------------------------------------
bitarray version: 2.9.2
bitstring version: 4.2.3
CacheControl version: 0.14.0
certifi version: 2024.2.2
cffi version: 1.16.0
charset-normalizer version: 3.3.2
click version: 8.0.4
colorama version: 0.4.6
construct version: 2.10.70
contextlib2 version: 21.6.0
cryptography version: 42.0.7
ecdsa version: 0.19.0
esp-coredump version: 1.11.0
esp-debug-backend version: 1.0.3
esp-idf-kconfig version: 2.2.0
esp-idf-monitor version: 1.4.0
esp_idf_nvs_partition_gen version: 0.1.2
esp-idf-panic-decoder version: 1.1.0
esp-idf-size version: 1.4.0
esptool version: 4.8.dev3
filelock version: 3.14.0
freertos-gdb version: 1.0.3
idf-component-manager version: 1.5.2
idna version: 3.7
intelhex version: 2.3.0
markdown-it-py version: 3.0.0
mdurl version: 0.1.2
msgpack version: 1.0.8
packaging version: 24.0
pip version: 24.0
pyclang version: 0.4.2
pycparser version: 2.22
pyelftools version: 0.31
pygdbmi version: 0.9.0.2
Pygments version: 2.18.0
pyparsing version: 3.1.2
pyserial version: 3.5
PyYAML version: 6.0.1
reedsolo version: 1.7.0
requests version: 2.32.2
requests-file version: 1.5.1
requests-toolbelt version: 1.0.0
rich version: 13.7.1
schema version: 0.7.7
setuptools version: 70.0.0
six version: 1.16.0
tqdm version: 4.66.4
urllib3 version: 1.26.18
---------------------------------------------------- Check ESP-IDF python requirements.txt -------------------------------------------------
Check ESP-IDF Python packages Python requirements are satisfied.
---------------------------------------------------- Check ESP-IDF debug adapter requirements.txt ------------------------------------------
Check Debug AdapterPython packages Python requirements are satisfied.
---------------------------------------------------- Visual Studio Code launch.json --------------------------------------------------------
{
  "configurations": [
    {
      "type": "gdb",
      "request": "attach",
      "name": "NativeDebug",
      "target": "extended-remote :3333",
      "executable": "${workspaceFolder}/build/${command:espIdf.getProjectName}.elf",
      "gdbpath": "${command:espIdf.getToolchainGdb}",
      "cwd": "${workspaceRoot}",
      "autorun": [
        "mon reset halt",
        "maintenance flush register-cache",
        "thb app_main"
      ]
    }
  ]
}
 
---------------------------------------------------- Visual Studio Code c_cpp_properties.json ----------------------------------------------
{
  "configurations": [
    {
      "name": "ESP-IDF",
      "compilerPath": "${config:idf.toolsPath}/tools/xtensa-esp-elf/esp-13.2.0_20240305/xtensa-esp-elf/bin/xtensa-esp32-elf-gcc",
      "compileCommands": "${workspaceFolder}/build/compile_commands.json",
      "includePath": [
        "${config:idf.espIdfPath}/components/**",
        "${config:idf.espIdfPathWin}/components/**",
        "${config:idf.espAdfPath}/components/**",
        "${config:idf.espAdfPathWin}/components/**",
        "${workspaceFolder}/**"
      ],
      "browse": {
        "path": [
          "${config:idf.espIdfPath}/components",
          "${config:idf.espIdfPathWin}/components",
          "${config:idf.espAdfPath}/components/**",
          "${config:idf.espAdfPathWin}/components/**",
          "${workspaceFolder}"
        ],
        "limitSymbolsToIncludedHeaders": false
      }
    }
  ],
  "version": 4
}
 
--------------------------------------------------------------------------------------------------------------------------------------------

### Description

I am using the following configuration for debugging:

```json
{
  "configurations": [
    {
      "type": "gdb",
      "request": "attach",
      "name": "NativeDebug",
      "target": "extended-remote :3333",
      "executable": "${workspaceFolder}/build/${command:espIdf.getProjectName}.elf",
      "gdbpath": "${command:espIdf.getToolchainGdb}",
      "cwd": "${workspaceRoot}",
      "autorun": [
        "mon reset halt",
        "maintenance flush register-cache",
        "thb app_main"
      ]
    }
  ]
}
```

In the above configuration `${command:espIdf.getToolchainGdb}` seems to be undefined. When running I receive the message:

![image](https://github.com/espressif/vscode-esp-idf-extension/assets/298182/c14d272d-dacd-46b5-9d89-b9601072a33d)

When I set the path manually as an absolute path, e.g.:

```bash
/Users/mczenko/.espressif/tools/xtensa-esp-elf-gdb/14.2_20240403/xtensa-esp-elf-gdb/bin/xtensa-esp32-elf-gdb
```

then I get the debugger working. Still getting a warning at the very beginning:

```
Could not get threads: Cannot execute this command while the target is running. Use the "interrupt" command to stop the target and then try again. (from thread-info)
```

yet, it seems to be working jus fine. I am also not able to restart the target - need to start new debugging session.
Finally, the `executable` setting in the launch configuration above does not seem to matter - I can change it to anything, remove, etc, and I get the same debugging output...

I am using `az-delivery-devkit-v4` with `Olimex arm-usb-tiny-h` interface.

### Debug Message

```plain
Warning: 'set target-async', an alias for the command 'set mi-async', is deprecated.
Use 'set mi-async'.

warning: No executable has been specified and target does not support
determining executable automatically.  Try using the "file" command.

0x40000400 in ?? ()
JTAG tap: esp32.cpu0 tap/device found: 0x120034e5 (mfg: 0x272 (Tensilica), part: 0x2003, ver: 0x1)
JTAG tap: esp32.cpu1 tap/device found: 0x120034e5 (mfg: 0x272 (Tensilica), part: 0x2003, ver: 0x1)
[esp32.cpu0] requesting target halt and executing a soft reset
[esp32.cpu0] Debug controller was reset.
[esp32.cpu0] Core was reset.
[esp32.cpu0] Target halted, PC=0x500000CF, debug_reason=00000000
[esp32.cpu1] Debug controller was reset.
[esp32.cpu1] Core was reset.
[[esp32.cpu1] Reset cause (14) - (CPU1 reset by CPU0)
[esp32.cpu1] requesting target halt and executing a soft reset
[esp32.cpu0] Core was reset.
[esp32.cpu0] Target halted, PC=0x40000400, debug_reason=00000000
[esp32.cpu1] Core was reset.
[esp32.cpu1] Target halted, PC=0x40000400, debug_reason=00000000
[esp32.cpu1] Reset cause (14) - (CPU1 reset by CPU0)
[esp32.cpu0] Reset cause (3) - (Software core reset)
Register cache flushed.
Hardware assisted breakpoint 1 at 0x400d6370: file /Users/mczenko/code/embedded/native-esp-idf/example-native-esp-idf/main/gpio_example_main.c, line 82.
[esp32.cpu0] Target halted, PC=0x400D6370, debug_reason=00000001
Set GDB target to 'esp32.cpu0'
[esp32.cpu1] Target halted, PC=0x4008444A, debug_reason=00000000
[New Thread 1073412648]
[Remote target exited]
[New Thread 1073413336]
[New Thread 1073412992]
[New Thread 1073412272]
[New Thread 1073410900]
[Switching to Thread 1073412648]
Thread 2 "main" hit Temporary breakpoint 1, app_main () at /Users/mczenko/code/embedded/native-esp-idf/example-native-esp-idf/main/gpio_example_main.c:82
82	{
```

### I have checked existing issues, online documentation and the Troubleshooting Guide

- [X] I confirm I have checked existing issues, online documentation and Troubleshooting guide.

**Comment by TixoRebel** [comment]
I see this as well on linux

**Comment by brianignacio5** [comment]
Are you using the extension from marketplace or the current master vsix ? We haven't release these changes to marketplace yet.

**Comment by marcinczenko** [comment]
I am using the extension from the marketplace.

**Comment by brianignacio5** [comment]
Please try the master latest installer in [here](https://nightly.link/espressif/vscode-esp-idf-extension/workflows/ci/master/esp-idf-extension.vsix.zip).

We will release soon but we need to do extension testing before release new version.

**Comment by brianignacio5** [comment]
Closing this issue since it is fixed in master and will be part of v1.8.0
